### PR TITLE
Trim down mutations section

### LINKED ISSUE #203: Contrast sim_mutations performance with existing methods
Reviewer 2 says:

> As another example, line 301-4 states that "Simulating mutations ... is efficient" and refers to Fig 3 for evidence. The rest of the section (304-319) details a number of examples shown in Fig 3 that do not add much to the story.

We originally avoided contrasting the performance of sim_mutations with Pyvolve and SeqGen because it's hard to disentangle the method performance from inefficient file formats. The concrete examples were intended to sidestep this by showing that sim_ancestry is as fast as you'd ever need it to be, but this point was lost I think. So, better to use the space to run comparable mutation simulations in Pyvolve and SeqGen and report how long they took.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Closes #203

I originally thought we needed to be more concrete about how we relate to other methods, but reading reviewer 2's comments again I see we just need to cut things down a bit.

I made another honest effort to get phastsim to work, but I just couldn't figure out how to make it simulate one of models we support. :shrug:

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -691,9 +691,7 @@ \subsection*{Simulating mutations}
 component intended to fit into a larger software ecosystem, and
 is in no way dependent on \msprime's ancestry simulator.
 
-As well as providing a new API
-that emphasises the logical split between ancestry and mutation simulation,
-we have greatly extended the sophistication of
+We have greatly extended the sophistication of
```
Comment (andrewkern) [inline paper.tex:694]: i prefer the old version here

File: paper.tex [hunk 2]
```diff
@@ -741,29 +739,17 @@ \subsection*{Simulating mutations}
 Fig.~\ref{fig-mutations-perf}B required less than 2 seconds to
 generate an average of 1.5 million mutations over 137,081 trees
 in a tree sequence with 508,125 edges.
-This efficiency for large numbers of trees is possible because
-the tree sequence encoding allows us to generate mutations
-on an edge-by-edge basis
-(see Fig.~\ref{fig-ts-example} and the~\nameref{app-mutation-algorithm}
-appendix),
-rather than tree-by-tree and branch-by-branch as would otherwise be required.
```
Comment (andrewkern) [inline paper.tex:749]: i wonder if this idea could be kept in the new version

**Review by andrewkern — COMMENTED** [review]
few minor comments

File: paper.tex (hunk 1 above)
Comment (jeromekelleher) [reply paper.tex:694]: Reviewer X was complaining that we kept talking about the API all the time, which was valid I think? Most readers won't care about this point and we want to shed some text

File: paper.tex (hunk 2 above)
Comment (jeromekelleher) [reply paper.tex:749]: It's said in a different way in the appendix (not quite as explicitly though, all right)

Comment (jeromekelleher) [reply paper.tex:749]: Although yeah, I see what you mean, it's independent of the following stuff. I'll have another look, ta.

**Pull request merged**